test/Npgsql.Tests/ReaderTests.cs

```diff
@@ -1228,6 +1228,62 @@ public async Task ReaderReuseOnDispose()
             Assert.That(reader1, Is.SameAs(reader2));
             await reader2.DisposeAsync();
         }
+
+        [Test]
+        public async Task UnboundReaderReuse()
+        {
+            var csb = new NpgsqlConnectionStringBuilder(ConnectionString)
+            {
+                MinPoolSize = 1,
+                MaxPoolSize = 1,
+            };
+
+
+            await using var conn1 = await OpenConnectionAsync(csb.ToString());
+            using var cmd1 = conn1.CreateCommand();
+            cmd1.CommandText = "SELECT 1";
+            var reader1 = await cmd1.ExecuteReaderAsync(Behavior);
+            Assert.That(async () => await reader1.ReadAsync(), Is.EqualTo(true));
+            Assert.That(() => reader1.GetInt32(0), Is.EqualTo(1));
+
+            var conn1ID = conn1.ProcessID;
+
+            await reader1.CloseAsync();
```

Comment:
Any reason not to use `await using` for all three of these?

Reply:
But if we do so, they will only be disposed when the test has ended, and so we cannot reuse them.

Reply:
But you can use the older syntax to dispose them wherever you want:

```c#
await using (var x = y) {

}
```